### PR TITLE
[Dashboard/SDK] Add more Split extensions and remove Split v4 code from Dashboard

### PR DESCRIPTION
**Pull request by kien-ngo**
## PR-Codex overview
This PR focuses on adding more functionality to the Split contract extensions, including deploying Split contracts, retrieving recipient addresses and percentages, and updating UI components.

### Detailed summary
- Added more Split contract extensions for deploying and reading data
- Updated UI components to display Split contract information
- Added tests for deploying Split contracts and retrieving recipient data

> The following files were skipped due to too many changes: `packages/thirdweb/src/extensions/split/__generated__/Split/read/totalShares.ts`, `apps/dashboard/src/contract-ui/tabs/split/components/distribute-button.tsx`, `packages/thirdweb/test/src/test-contracts.ts`, `apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts`, `packages/thirdweb/src/extensions/prebuilts/deploy-split.ts`, `packages/thirdweb/src/extensions/prebuilts/__generated__/Split/write/initialize.ts`, `apps/dashboard/src/contract-ui/tabs/split/page.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts [hunk 1]
```diff
@@ -48,72 +40,30 @@ export function useSplitBalances(contractAddress?: string) {
...
+                  contract,
+                  tokenAddress: currency.token_address,
                 });
-              });
-          }
+          await mutateAsync(transaction)
+            .then(() =>
+              toast.success(`Successfully distributed ${currency.name}`),
+            )
+            .catch(() => toast.error(`Error distributing ${currency.name}`));
```
Comment (jnsdls) [inline apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts:66]: consider using `toast.promise` here: https://sonner.emilkowal.ski/toast#promise
```
const prom = mutateAsync(transaction)
toast.promise(prom, ...);
...

File: packages/thirdweb/test/src/test-contracts.ts [hunk 2]
```diff
@@ -1,68 +1,84 @@
...
+
+/**
+ * All access to this contract (admin roles) have been removed
+ * so this contract is effectively read-only
+ *
+ * cons: We might have to update the test if avalancheFuji is deprecated
+ *
+ * link: https://thirdweb.com/avalanche-fuji/0x87077BDEEC6Bc1Bc0D399c76742b49A5844F66F8/permissions
+ */
+export const TEST_SPLIT_CONTRACT = getContract({
```
Comment (jnsdls) [inline packages/thirdweb/test/src/test-contracts.ts:80]: can we instead deploy this onto the local testnet to test it - relying on network calls to test seems potentially problematic

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts:66]: woa didn't know about this

File: packages/thirdweb/test/src/test-contracts.ts (hunk 2 above)
Comment (kien-ngo) [reply packages/thirdweb/test/src/test-contracts.ts:80]: Ok working on a deploy-split extension now.

I kinda hoped you wouldn't notice this lmaoo (jk)

Comment (kien-ngo) [reply packages/thirdweb/test/src/test-contracts.ts]: I was gonna replace the Split contract deployment code to v5, in this PR however I feel like it's gonna take a bit more time and I just want this PR out first

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts [hunk 3]
```diff
@@ -48,72 +40,30 @@ export function useSplitBalances(contractAddress?: string) {
...
+            currency.name === "Native Token"
+              ? distribute({ contract })
+              : distributeByToken({
+                  contract,
+                  tokenAddress: currency.token_address,
                 });
-              });
-          }
+          const promise = mutateAsync(transaction);
+          toast.promise(promise, {
```
Comment (jnsdls) [inline apps/dashboard/src/@3rdweb-sdk/react/hooks/useSplit.ts:63]: I believe you can give this a `loading` text, too - which is what it'll show while the promise is executing

**Review by jnsdls — CHANGES_REQUESTED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**